### ISSUE TITLE
Not able to import from google spreadsheet

### ISSUE THREAD

**Issue by carma439** [ThePieMonster/WaniKanify-2.0#11]
Hello,
I want to import my list but have been unable to do so even after following the steps mentioned.
My published spreadsheet link:
https://docs.google.com/spreadsheets/d/e/2PACX-1vSpnM_VBwJAgjn9o50kKBhDWwthoJPe7pTXeQBeOleussgO2R2NqfEB36qFf2uOCquyxOJxkz4tumjW/pubhtml

Values I've entered in the extensions options:

1. key: 2PACX-1vSpnM_VBwJAgjn9o50kKBhDWwthoJPe7pTXeQBeOleussgO2R2NqfEB36qFf2uOCquyxOJxkz4tumjW
2. From column header: English
3. Delimiter: comma (,)
4. To column header: Kanji
5. Reading column: -blank- didn't include readings
6. Sheet Name: Sheet1

Still, it says import failed everytime I try.
Please help. : (I'm new to Github and this technical stuff)

**Comment by ThePieMonster** [comment]
I see based on your WaniKani level and the information you are trying to import, you want to use WaniKanify 2.0 without actually using WaniKani. All of the Kanji that you are trying to import are in WaniKani already and are displayed when you reach the appropriate level. 

As to why the Excel import is failing, you are the first to really use that feature as not even myself continues to work with Google and its import process due to the numerous issues and time commitments required. 

It would be in your best interest to do one of the following:

- Use WaniKani and unlock the Kanji as intended by the service, thus allowing WaniKanify 2.0 to pull the data for your level
- Manually use the text delimited list instead of the Excel import feature, thus you can use WaniKanify 2.0 without actually using WaniKani

**Comment by carma439** [comment]
Yes, I wanted to use Wanikanify without using Wanikani, since I'm not able to afford Wanikani currently.
So, I learnt top 300 kanji by frequency first (as you can see in the spreadsheet).
The Excel import feature is really amazing (if it works xd).

By using text delimited list, you mean the 'custom vocab list'?
Currently, I'm using that as the list contains about 300 items and it works very well.
However, Excel works much better in the long run as it is way easier to update the list as I continue my studies.
For eg. I wrote 
year; 年 
in the custom vocab list. But it didn't change the word 'years', so I had to add years as a separate keyword and then it worked.
But updating the custom vocab list is not that good compared to updating excel sheet.
(because you need to make sure you are in the same line for the kanji and words)

Also, I guess, excel import feature automatically updates the list in Wanikanify as I keep updating the Excel sheet. This would have been a lot easier.

I wish I could buy Wanikani, but the steep price and learning kanji readings separately seems counter intuitive to me.

By the way, have you heard of Toucan extension.
It changes some random words into the language you are studying.
It works very well for context mapping.
I wish there was some way to combine the custom list/ excel import feature of Wanikanify with Toucan.
Thank you!

**Comment by ThePieMonster** [comment]
Thanks for the response. Also I wanted to mention that this project is open to contributions from all members on GitHub. If you feel you want to add a feature or resolve any bugs you are free to do so and merge your changes into the project for other members to benefit from. 

Since the Excel import is having issues, I would look to convert your Excel sheet to a delimited list, with each item being separated with a semicolon as shown in the example "custom vocab" of the extension. This will allow you to use the extension in the manner that you desire. You do have to be specific with both the Excel and custom vocab sections, "years" is not the same as "year" although in our human minds it is, to a computer it is not. Perhaps another feature for the future would be to match like terms even if they do end in a 's'. So "year" matches "year" and "years". 

I will keep this ticket open due the Excel import feature having issues.

**Comment by carma439** [comment]
Thank you!
Though as I mentioned above, I am not a programmer.
So, I have no idea how do you create extensions, use Wanikani APIs or how does the excel import feature actually works.

That said, I'm eager to learn the programming stuff required to work on this extension.
This extension uses Javascript and I only know basics of Python and HTML.
Can you suggest some resources from where should I learn about Javascript and the other required stuff to work on this extension?

I would like to make this extension more helpful to others and me!

**Comment by ThePieMonster** [comment]
No problem! We all start off as non-developers and work our way up. My comments are open for anyone to see here and wanted to inform anyone else the state of this bug. In addition you can reach out to me over Discord, information in my profile. Don't want to overpopulate a issue thread with non-related information but to answer your question see the below.

In regards to recommended resources everyone finds something different to be "more helpful". That being said, I think the below would be a fair start, and you can fork this repo using Git and then test changes locally. This project is fairly straightforward, with the api calls, dictionary modifications, and dealing with google import being the most complicated aspects. So not rocket science but you will have to spend a little time before understanding. Also enrolling in a actual graded course might be useful in motivational terms. 

[Mozilla Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide)
[WebTips - Chrome Extension](https://www.webtips.dev/how-to-make-your-very-first-chrome-extension)
[Git Extensions](http://gitextensions.github.io/)
